src/main/java/app/controller/impl/ReviewControllerImpl.java

```diff
@@ -0,0 +1,132 @@
+package app.controller.impl;
+
+import app.controller.Controller;
+import app.dao.impl.ReviewDAOImpl;
+import app.dto.ReviewDTO;
+import app.entity.Review;
+import app.mapper.ReviewMapper;
+import io.javalin.http.*;
+import io.javalin.validation.ValidationException;
+import jakarta.persistence.EntityExistsException;
+import jakarta.persistence.EntityManagerFactory;
+import jakarta.persistence.EntityNotFoundException;
+import org.hibernate.MappingException;
+
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class ReviewControllerImpl implements Controller {
+
+    private static ReviewControllerImpl instance;
+
+    private final ReviewDAOImpl reviewDAO;
+    private final ReviewMapper reviewMapper;
+
+    private ReviewControllerImpl(EntityManagerFactory emf) {
+        this.reviewDAO = ReviewDAOImpl.getInstance(emf);
+        this.reviewMapper = new ReviewMapper();
+    }
+
+    public static ReviewControllerImpl getInstance(EntityManagerFactory emf) {
+        if (instance == null) {
+            instance = new ReviewControllerImpl(emf);
+        }
+
+        return instance;
+    }
+
+    @Override
+    public void getById(Context ctx) {
+        try {
+            Long id = ctx.pathParamAsClass("id", Long.class).get();
+
+            Review foundReview = reviewDAO.getById(id);
+            ReviewDTO mappedReviewDTO = reviewMapper.convertToDTO(foundReview);
+
+            ctx.status(HttpStatus.OK);
+            ctx.json(mappedReviewDTO, ReviewDTO.class);
+        } catch (ValidationException e) {
+            throw new BadRequestResponse(e.getErrors().toString());
+        } catch (MappingException e) {
+            throw new BadRequestResponse(e.getMessage());
+        } catch (EntityNotFoundException e) {
+            throw new NotFoundResponse(e.getMessage());
+        }
+    }
+
+    @Override
+    public void getAll(Context ctx) {
+        try {
+            String idParam = ctx.queryParam("id");
+            Long id = Long.valueOf(idParam);
```

Comment:
This should be changed to use `ctx.queryParamAsClass()` so it will be properly validated as users can potentially call this route with an incorrect gameId query parameter